src/ripple/basics/base_uint.h

```diff
@@ -37,6 +37,20 @@
 
 namespace ripple {
 
+namespace detail
+{
+
+template <class Container>
+using contiguous_container
+    = std::void_t
+      <
+          decltype(std::declval<Container const>().size()),
+          decltype(std::declval<Container const>().data()),
+          typename Container::value_type
+      >;
```

Comment:
Nice.  Just to prove to myself that this concept works, I tried this:
```
    {
        uint128 b;
        std::list<std::uint8_t> l (sizeof (b));
        b = l;
    }
```
The compiler diagnostic I got back (from clang) was:
```
no member named 'data' in 'std::__1::list<unsigned char...
```
Thanks for the education.